tools/autograd/load_derivatives.py

```diff
@@ -92,34 +92,40 @@ def create_derivative(f: NativeFunction, formula: str, var_names: Tuple[str, ...
     )
 
 def create_forward_derivative(f: NativeFunction, formula: str, names: Tuple[str, ...]) -> ForwardDerivative:
-    assert len(names) == 1, "Forward derivatives can define gradients for only one output at a time"
-    var_name = names[0]
-    var_type: Optional[Type] = None
+    var_names = names
+    var_types: Optional[Tuple[Type, ...]] = None
     for r in f.func.returns:
-        if r.name == var_name:
-            var_type = r.type
-            break
+        if r.name in var_names:
+            if var_types is None:
+                var_types = tuple()
+            var_types = var_types + (r.type,)
+
     # Handle default return names
-    if var_type is None:
-        if var_name == "result":
+    if var_types is None:
+        if var_names == ("result",):
             assert len(f.func.returns) == 1
-            var_type = f.func.returns[0].type
+            var_types = (f.func.returns[0].type,)
         else:
-            res = re.findall(r"^result(\d+)$", var_name)
-            if len(res) == 1:
-                arg_idx = int(res[0])
-                var_type = f.func.returns[arg_idx].type
-
-    assert var_type is not None, "No matching output for forward derivative definition"
-    return ForwardDerivative(
+            for var_name in var_names:
+                res = re.findall(r"^result(\d+)$", var_name)
+                if len(res) == 1:
+                    if var_types is None:
+                        var_types = tuple()
+                    arg_idx = int(res[0])
+                    var_types = var_types + (f.func.returns[arg_idx].type,)
+
+    assert var_types is not None, "No matching output for forward derivative definition"
+    fwd_deriv = ForwardDerivative(
         formula=formula,
-        var_name=var_name,
-        var_type=var_type,
+        var_names=var_names,
+        var_types=var_types,
         required_inputs_fw_grad=None,
         required_inputs_primal=None,
         required_original_self_value=False,
         is_reusing_outplace_formula=False)
 
+    return fwd_deriv
```

Comment:
Is there any reason to move this here?

Reply:
Nope, just wanted to print it at one point